npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/yarn_lockfile_updater.rb

```diff
@@ -380,9 +380,10 @@ def clean_npmrc_in_path(yarn_lock)
           dirs.pop
           while dirs.any?
             npmrc = dirs.join("/") + "/.npmrc"
-            break unless File.exist?(npmrc)
```

Comment:
I assume this was breaking and not walking through the entire dirs right?

Reply:
Yes. If the subdir did not contain the `.npmrc` file then it was breaking out. In such case if the repo root dir still had `.npmrc` file then error was getting raised.